### PR TITLE
feat(notification): add Python runtime version information to device data

### PR DESCRIPTION
**Pull request by tomlongridge**
adds version string (from platform.python_version()) to device.runtimeVersions.python payload

## Goal

Augments reporting with the version of Python in use when an event occurred in case of issues in a specific Python version.

## Design

Added data to device.runtimeVersions.python as per other notifiers.

### Changed

* Added new field in configuration, set to `platform.python_version()` on init
* `notification` and `sessiontracker` use this data in their payloads

## Tests

* Added unit tests for the presence of the data
* Confirmed data is there in example project

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: bugsnag/configuration.py [hunk 1]
```diff
@@ -92,6 +93,8 @@ def __init__(self):
         else:
             self.hostname = None
 
+        self.runtime_versions = {"python": str(platform.python_version())}
```
[the remark this answers is not in the document]
Comment (tomlongridge) [reply bugsnag/configuration.py]: Ah, I misread the [docs](https://docs.python.org/2/library/platform.html) when it said defaults to 0 for patch version.

TBH, it was an optimistic addition to try and resolve the build failures. I thought it was happening on Travis for py34 only, but the build history seems to suggest these failures are randomly happening but only on this branch.

They are all `TypeError: string indices must be integers` errors when examining the payload. However I cannot see how adding a new field to the dict should cause this. I might need to dig into the FilterDict code to see if this is somehow causing some unexpected changes to the structure.

[the remark this answers is not in the document]
Comment (tobyhs) [reply bugsnag/configuration.py]: I can't tell if you were planning on removing the `str` call or not.
